### PR TITLE
Add FAQ page

### PR DESCRIPTION
**Pull request by Adrianzatreanu**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -51,6 +51,9 @@ To see what coala can do for you and your language, take a look at
 
 Read more at our `documentation <http://coala.rtfd.org/>`__.
 
+To consult our list of Frequently Asked Questions, simply visit `this link <http://coala.readthedocs.io/en/latest/Users/FAQ.html>`__
+.
```
Comment (sils1297) [inline README.rst]: period in an own line? :/

Comment (Adrianzatreanu) [inline README.rst]: if i'd add it on the same line as before, wouldnt it count as linelength > 80?

File: docs/Users/FAQ.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+list of how to get in touch with us, consult `this link <http://coala.readthedocs.io/en/latest/Getting_Involved/README.html>`__.
+
+What are those things failing/passing on my Pull Request?
+=========================================================
+
+We use a few checks to make sure your Pull Request is ready to be merged into
+our master branch. Right now we use 7 of those checks:
+
+- **review/gitmate/commit** Checks whether your commit message is alright and
+  doesn't have too many characters per line.
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: nope, it's if there are any new results in coala for this particular commit

File: docs/Users/FAQ.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+- **review/gitmate/pr** Checks whether your code respects our styling (PEP8),
+  doesn't contain unneeded newlines, trailing whitespace, etc. Basically it is
+  like running coala over your code, so to fix this, simply run ``$ coala``
+  before pushing!
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: the gitmate PR state will fail if any of the commit states fail so if the last comit is good but the previous ones aren't we want the PR to still have a failed state

File: docs/Users/FAQ.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+- **review/gitmate/manual** This one is the only one that is manual, this can be
+  given by any coala member and shows that the commit has been reviewed and
+  has no problems, so it is ready to be merged. It can be done by commenting
+  ``ack commit_sha``.
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: add a link to our review documentation, there's some elaborate explanantion there

File: docs/Users/FAQ.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+- **scrutinizer** Checks for the code elements that were added.
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: this description isn't really helpful to me at least

File: docs/Users/FAQ.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+- **codecov/project** This one checks whether all your code is being tested. We
+  cannot merge anything that may not work or may broke somewhere, so to avoid
+  obvious bugs we use this. To fix it, write doctests or unittests for your
+  functions / classes.
+
+- **ci/circleCI** This is one of the two containers we use to continuously
+  integrate the code. It basically runs all the tests and checks your code in a
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: we integrate code? :P Can we go with test instead?

File: docs/Users/FAQ.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+coala was founded for fun. coala was, is and will always be free software and is
+developed mostly by students and there's no corporate interest and no CLA. If
+you want to back us, contact us on our `gitter channel <https://gitter.im/coala-analyzer/coala>`__
+.
+
+What sort of analysis does coala do? What languages are supported?
+==================================================================
+
+A list of all analysis routines and supported languages is `here <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst#supported-languages>`__
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: I'd break links before the backtick in general, can you change that for this commit?

Comment (sils1297) [inline docs/Users/FAQ.rst]: that is if the line grows beyond 80 chars of course

File: docs/index.rst [hunk 8]
```diff
@@ -51,7 +51,7 @@ Welcome to the coala documentation!
    :hidden:
 
    Newcomers <Getting_Involved/Newcomers>
-   Introduction <Getting_Involved/README>
+   How to get in touch with us <Getting_Involved/README>
```
Comment (sils1297) [inline docs/index.rst]: can you use title caps?

File: docs/Users/FAQ.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+What sort of analysis does coala do? What languages are supported?
+==================================================================
+
+A list of all analysis routines and supported languages is
+`here <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst#supported-languages>`__
+- fully browsable.
+
+For a top level view on what languages support what kind of analysis roughly,
+consult `this link <https://docs.google.com/spreadsheets/d/1bm63TQHndmGf3HQ33fp9UEmGKNYI7dTkjMyFIof2PqA/edit?usp=sharing>`__.
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: maybe add a paragraph about the generic stuff?

File: docs/Users/FAQ.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+`here <http://coala.readthedocs.io/en/latest/Getting_Involved/Newcomers.html>`__.
+It is meant for newcomers, and it does not require you to have any precedent
+knowledge regarding coala.
+
+How do I get in touch with you guys?
+====================================
+
+We are very active on our
+`gitter channel <https://gitter.im/coala-analyzer/coala>`__
+and will respond to any question in a matter of minutes. However, for a full
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: `we try to respond ...`

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2428/commits/31486e0c8175ba86e799d57818608cb253baf056 https://github.com/coala-analyzer/coala/pull/2428/commits/7fb50b106ba8ebb892f0b746f765c2e644916b79 (like this :)) https://github.com/coala-analyzer/coala/pull/2428/commits/de04cb6ae6b2f6916f590ec048a3360c6d1ef96e

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/2428/commits/b0b2b691d4a352c069b85bd72adb5927debe44fa needs work

**Comment by Adrianzatreanu** [comment]
reack 6fdf392 866a247 5770fe8

File: docs/Users/FAQ.rst [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+There's also generic bears, which can be applied language independently on your
+code:
+
+- **FilenameBear** Checks the filenames of the files in your project.
+- **InvalidLinkBear** Checks for invalid links in your code.
+- **KeywordBear** Checks for a given keyword in your code.
+- **LineCountBear** Counts the number of lines in your project
+- **LineLengthBear** Checks whether you have lines over (default) 80 characters.
+- **SpaceConsistencyBear** Checks and fixes if your code contains mixed spaces
+  and tabs, and uses only the one you specify.
```
Comment (sils1297) [inline docs/Users/FAQ.rst]: don't list them here, this'll get outdated, link to the bear-docs all section

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/2428/commits/3aff83d655e68d6d6749d16b54dcb821a7a74f6a unack

**Comment by Adrianzatreanu** [comment]
reack
1ceab32
 @Adrianzatreanu	Git_Help: Add rebase video			89adc88
 @Adrianzatreanu	Git_Help.rst: Fix grammar problem			1e6b4e4

**Comment by sils1297** [comment]
reack
1ceab32
@Adrianzatreanu Git_Help: Add rebase video 89adc88
@Adrianzatreanu Git_Help.rst: Fix grammar problem 1e6b4e4

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2428/commits/20ac7bf2cfa2ea633e275a45d13dc386368f2892  
https://github.com/coala-analyzer/coala/pull/2428/commits/1ceab32e9ea848672920851cf590df5fc389aa82 
https://github.com/coala-analyzer/coala/pull/2428/commits/89adc881c94be65167a3f87abced63c9b80fe180 
https://github.com/coala-analyzer/coala/pull/2428/commits/1e6b4e4d07c16964193b6e7304c20a8459782d85

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Pull request merged**